scripts/beautify.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# Reformat C++ code using clang-format
+
+# note: with the -style=file option, clang-format reads the config from ./.clang-format
+# See here for the full list of supported options: https://clang.llvm.org/docs/ClangFormatStyleOptions.html
+FORMAT_CMD="clang-format -i -style=file"
+
+function reformat_all() {
+    find . -name "*.h" -o -name "*.cc" | xargs $FORMAT_CMD
```

Comment:
We might want to add "*.hpp", "*.cpp" to this.
Also note that some code originates elsewhere and keeping the diff small is probably more valuable than coding style (e.g. `polyclipping/`, `CGAL_Nef3_workaround.h`, `CGAL_workaround_Mark_bounded_volumes.h`, `convex_hull_3_bugfix.h`, `OGL_helper.h`

Reply:
Good point, I went ahead and added those extensions.

The simplest way to exclude files if there are just a few would probably be to add `// clang-format off`/on comments them. I could also add a file blacklist to the script, but it would add a bit of complexity to get it to work with both `reformat_all` and `reformat_changed`.

Reply:
The CGAL files could easily get `// clang-format *` added, but the polyclipping ones are verbatimely copied from upstream. Could we ignore that entire folder somehow?

Reply:
Alternatively, we could move the folder out of the `src/` into a new `ext/` folder or smth.

Reply:
I ended up finding a relatively simple way to exclude a list of file patterns, so I think we can do that instead of having to modify or move those files.